ckan/logic/action/patch.py

```diff
@@ -69,6 +70,7 @@ def resource_patch(context, data_dict):
         'session': context['session'],
         'user': context['user'],
         'auth_user_obj': context['auth_user_obj'],
+        'for_update': True
```

Comment:
resource_show attaches a resource model to the context before calling package_show with for_update in the context, so the resource object in our context might differ from the one returned by package_show. I guess this doesn't matter because we throw away the show_context, but it's the kind of thing I can see breaking as changes are made in the future.

I think some tests would be the best way to mitigate this and cover this all the new functionality in general. Can you think of a way to test this feature?

Reply:
What sort of tests would we need here @wardi? a test with concurrent resource updates? or that reuse the same context between `resource_patch` calls?

Reply:
kind of tricky to write a test of concurrent updates actually being serialized by postgres. Maybe good enough is a test that verifies the `for_update==True` reaches the dictization layer.

Reply:
Wardi, can you give some pointers so I can start making the test 
Thanks

Reply:
@wardi I tried to write an example for Tome but I'm a bit lost with how the `for_update` flag works. You mention the dictization layer but AFAICT this never reaches that level and it is instead used at the [model level](https://github.com/ckan/ckan/blob/36839d57d443bd189056bc83cf3e684c72a1f5d5/ckan/model/package.py#L87). Is this what you meant?

Reply:
yes, I misremembered.

Reply:
@amercader you're right, the model level not dictization. I think we could mock that method just to verify that the actions do in fact pass the flag properly.